README.md

```diff
@@ -1,41 +1,60 @@
+[![Review Assignment Due Date](https://classroom.github.com/assets/deadline-readme-button-24ddc0f5d75046c5622901739e7c5dd533143b0c8e959d652212380cedb1ea36.svg)](https://classroom.github.com/a/-ucQIGTc)
+[![Open in Visual Studio Code](https://classroom.github.com/assets/open-in-vscode-718a45dd9cf7e7f842a935f5ebbe5719a5e09af4491e668f4dbf3b35d5cca122.svg)](https://classroom.github.com/online_ide?assignment_repo_id=15216106&assignment_repo_type=AssignmentRepo)
 # SE-Assignment-2
 Assignment: Introduction to Software Engineering
 Instructions:
 Answer the following questions based on your understanding of software engineering concepts. Provide detailed explanations and examples where appropriate.
 
 Questions:
 Define Software Engineering:
-
+Software engineering is the systematic approach to software development, operation, and maintenance. It includes applying engineering principles and practices during all stages of the software development process: requirements analysis, design, implementation, testing, deployment, and support of developed software. This is a way of producing properly working software and also answering user needs with high product quality, reliability, efficiency, and maintainability.
 What is software engineering, and how does it differ from traditional programming?
 Software Development Life Cycle (SDLC):
-
```

Comment:
The Software Development Life Cycle (SDLC) is a framework that describes the stages that are followed in the development of software. Phases usually include:
- Requirements Analysis: Gathering and defining the software requirements. 
- Design: Produce elaborated design of the software architecture, and components.
- Implementation: Writing the code and building the software according to the design.
- Testing: Testing to ensure that the software meets the requirements outlined and is free of any bugs.
- Deployment: Releasing the software for end-user use. 
- Maintenance: Modifying, updating, or enhancing the software as per the feedback and issues of the users.